### PR TITLE
[11.x] Allow to set a name when registering a blade extension

### PR DESCRIPTION
**Pull request by artrz**
This change allows to set a name when registering a blade extension, which can be used to override or control extensions added by third party packages.

### CODE REVIEW

File: src/Illuminate/View/Compilers/BladeCompiler.php [hunk 1]
```diff
@@ -679,13 +679,22 @@ public function stripParentheses($expression)
...
     /**
      * Register a custom Blade compiler.
-     *
-     * @param  callable  $compiler
-     * @return void
      */
-    public function extend(callable $compiler)
+    public function extend(callable $compiler, ?string $name = null): void
+    {
+        if (! empty($name)) {
```
Comment (rodrigopedra) [inline src/Illuminate/View/Compilers/BladeCompiler.php:685]: How about `if (filled($name)) {`?

`filled()` will trim the string before checking if it is not blank. And also check for non-null values.

**Comment by taylorotwell** [comment]
What is the benefit / use case?

**Comment by artrz** [comment]
@taylorotwell If packages extending the blade compiler like https://github.com/conedevelopment/blade-filters/blob/master/src/BladeFiltersServiceProvider.php could add a name it would be easy to override the registered callable.

File: src/Illuminate/View/Compilers/BladeCompiler.php (hunk 1 above)
Comment (artrz) [reply src/Illuminate/View/Compilers/BladeCompiler.php:685]: Thanks! I'll update if there's interest in merging the PR.

**Comment by driesvints** [comment]
@artrz did you need a new review? Make sure to mark the PR as ready if you need one.

File: src/Illuminate/View/Compilers/BladeCompiler.php (hunk 1 above)
Comment (artrz) [reply src/Illuminate/View/Compilers/BladeCompiler.php]: Updated.

**Comment by artrz** [comment]
Thanks for the heads-up @driesvints

**Comment by taylorotwell** [comment]
Thanks for your pull request to Laravel!

Unfortunately, I'm going to delay merging this code for now. To preserve our ability to adequately maintain the framework, we need to be very careful regarding the amount of code we include. 

If applicable, please consider releasing your code as a package so that the community can still take advantage of your contributions!

**Pull request closed without merging**